src/sdk/namespace_cache.js

```diff
@@ -582,7 +665,15 @@ class NamespaceCache {
 
             const hub_stream = new stream.PassThrough();
             const hub_params = { ...params, source_stream: hub_stream };
+            const start = process.hrtime.bigint();
             const hub_promise = this.namespace_hub.upload_object(hub_params, object_sdk);
+            // update latency stats on 'end'
+            hub_promise.once('end', () => {
+                this.stats_collector.update_hub_latency_stats({
+                bucket_name: params.bucket,
+                hub_write_latency: Number(process.hrtime.bigint() - start) / 1e6,
+                });
+            });
```

Comment:
for a promise we need to use .then like this:
```suggestion
            hub_promise.then(() => this.stats_collector.update_hub_latency_stats({
                bucket_name: params.bucket,
                hub_write_latency: Number(process.hrtime.bigint() - start) / 1e6,
            }));
```